api/envoy/type/matchers/metadata.proto

```diff
@@ -0,0 +1,59 @@
+syntax = "proto3";
+
+package envoy.type.matchers;
+option go_package = "matchers";
+
+import "envoy/type/matchers/string.proto";
+import "envoy/type/matchers/number.proto";
+
+import "validate/validate.proto";
+
+// [#protodoc-title: MetadataMatcher]
+
+// MetadataMatcher provides a general interface to check if a given value is matched in
+// :ref:`Metadata <envoy_api_msg_core.Metadata>`. It uses `filter` and `path` to retrieve the value
+// from the Metadata and then check if it's matched to one of the specified values.
+//
+// An example use of MetadataMatcher is specifying additional metadata in envoy.filters.http.rbac to
+// enforce access control based on dynamic metadata in a request.
+message MetadataMatcher {
+  // Specifies the value to match. Only primitive value is supported. For non-primitive value, the
+  // result is always not matched.
+  message Value {
+    // Specifies how to match a value. Only have effect on primitive value.
+    oneof match_pattern {
+      option (validate.required) = true;
+
+      // If specified, it's matched if and only if the target value is a NullValue and this field is
```

Comment:
what is the use case of setting null_match to `false`? (same for below)

Reply:
Not too much real use case, I make it a `bool` to simplify the `Value` design and also feel it's easier to set a bool field for the user.

Reply:
I would prefer using a empty message, like here:
https://github.com/envoyproxy/envoy/blob/master/api/envoy/api/v2/route/route.proto#L763

Reply:
Done.